debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/spi/Snapshotter.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.postgresql.spi;
+
+import io.debezium.connector.postgresql.PostgresConnectorConfig;
+import io.debezium.relational.TableId;
+
+import java.util.Optional;
+
+public interface Snapshotter {
+    void init(PostgresConnectorConfig config, OffsetState sourceInfo,
+              SlotState slotState);
+
+    boolean shouldSnapshot();
+    boolean shouldStream();
+
+    Optional<String> buildSnapshotQuery(TableId tableId);
```

Comment:
Do implementations need the database name actually? If not, I'd suggest to simply pass in table and schema name as two strings, thus avoiding the reference to `TableId`.

Reply:
The built-in snapshotters (via the `QueryingSnapshotter`) make use of the `TableId` class as that has utility methods for escaping the table names, etc. While it is exposing one more class, the TableId class is quite useful and has no internal concerns (AFAICT) associated with it